src/tokens/transformation/tailwind-email/formatTokens.ts

```diff
@@ -0,0 +1,159 @@
+ */
+function createColorTokensFromGroup(tokens, staticTheme = true) {
+  const colorTokens = {}
+  tokens.forEach(({ type, name, ...t }) => {
+    if (type === 'color') {
+      /**
+       * The following conditions are in order to properly group
+       * color tokens and format into a nested object structure
+       * for use in Tailwind.
+       */
+      let colorGroup = colorTokens[t.attributes.type] ?? {}
+
+      const tItem = kebabCase(t.attributes.item)
+      const tSubItem = kebabCase(t.attributes.subitem)
+
+      /**
+       * `state` is for the deepest level on a token.
+       * E.g. `icon` in colors.systemfeedback.success.icon
+       */
+      if (t.attributes.state) {
+        if (!staticTheme) {
+          // If not on a static theme, do not place within `dark` or `light` groups
+          colorTokens[tItem] = colorTokens[tItem] || {}
+          const tokenGroup = colorTokens[tItem][tSubItem] ?? {}
+          colorTokens[tItem][tSubItem] = merge(tokenGroup, {
+            [t.attributes.state]: t.value
+          })
+        } else {
+          // If on a static theme, place within `dark` or `light` groups
+          const tokenGroup = colorGroup[tItem]
+          colorGroup[tItem] = merge(tokenGroup, {
+            [tSubItem]: t.value
+          })
+        }
+      } else if (tSubItem) {
+        /**
+         * If not on a static theme AND theme is determined by `type`
+         * property do not place within `dark` or `light` groups
+         */
+        if (themes.includes(t.attributes.type) && !staticTheme) {
+          const tokenGroup = colorTokens[tItem] ?? {}
+          colorTokens[tItem] = merge(tokenGroup, {
+            [tSubItem]: t.value
+          })
+
+          /**
+           * If not on a static theme AND theme is determined by `item`
+           * property (e.g. legacy tokens) do not place within `dark`
+           * or `light` groups
+           */
+        } else if (themes.includes(t.attributes.item) && !staticTheme) {
+          const tokenGroup = colorTokens[t.attributes.type] ?? {}
+          colorTokens[t.attributes.type] = merge(tokenGroup, {
+            [tSubItem]: t.value
+          })
+        } else {
+          // If on a static theme, place within `dark` or `light` groups
+          const tokenGroup = colorGroup[tItem]
+          colorGroup[tItem] = merge(tokenGroup, {
+            [tSubItem]: t.value
+          })
+        }
+
+        /**
+         * If `item` property is the token name, don't nest inside object
+         */
+      } else if (t.attributes.item) {
+        colorGroup[tItem] = t.value
+
+        /**
+         * If `item` property is the token name, set directly on colorGroup
+         */
+      } else if (t.attributes.type) {
+        colorGroup = t.value
+      }
+
+      if (Object.keys(colorGroup).length > 0) {
+        colorTokens[t.attributes.type] = colorGroup
+      }
+    }
+  })
+  return colorTokens
+}
+
+export default (({ dictionary }) => {
```

Comment:
nit: Can we throw a type on dictionary?


```suggestion
export default (({ dictionary }: { dictionary: Dictionary }) => {
```